cgo_static.go

```diff
@@ -1,13 +1,15 @@
-//go:build !duckdb_use_lib && (darwin || (linux && (amd64 || arm64)) || (freebsd && amd64))
+//go:build !duckdb_use_lib && (darwin || (linux && (amd64 || arm64)) || (freebsd && amd64) || (windows && amd64))
 
 package duckdb
 
 /*
+#cgo CPPFLAGS: -DDUCKDB_STATIC
 #cgo LDFLAGS: -lduckdb
 #cgo darwin,amd64 LDFLAGS: -lc++ -L${SRCDIR}/deps/darwin_amd64
 #cgo darwin,arm64 LDFLAGS: -lc++ -L${SRCDIR}/deps/darwin_arm64
 #cgo linux,amd64 LDFLAGS: -lstdc++ -lm -ldl -L${SRCDIR}/deps/linux_amd64
 #cgo linux,arm64 LDFLAGS: -lstdc++ -lm -ldl -L${SRCDIR}/deps/linux_arm64
+#cgo windows,amd64 LDFLAGS: -lws2_32 -lwsock32 -lRstrtmgr -lstdc++ -lm --static -L${SRCDIR}/deps/windows_amd64
```

Comment:
Out of curiosity - how did you know which flags to add?
I found this. Did you add them by looking at the error messages about missing libraries?
> For static linking, one has to specify all the libraries explicitly, including those required indirectly (i.e. by libraries that you use).

Reply:
>  I found this. Did you add them by looking at the error messages about missing libraries?

Yes, exactly. I just looked at the error messages. Waiting half an hour every time wasn't ideal, but it worked :)

Reply:
Haha yes I can imagine 😄

Reply:
Hi @JAicewizard, thanks for your contribution btw . I'm trying to build for windows with [goreleaser](https://github.com/goreleaser/goreleaser-cross)  and it's failing with 

```
/usr/bin/x86_64-w64-mingw32-ld: cannot find -lRstrtmgr
````

Do you know how can one get the libraries needed to compile?

Reply:
If your ditro doesn't ship this library you're out of luck for cross-compiling.
But you could try https://github.com/marcboeker/go-duckdb/pull/293 which made everything lowercase

Reply:
Thanks a lot for your reply. Making the R lowercase took me to the next issue, which was adding `-lpthread`. now im fighting some std missing functions. Thanks a lot for your reply, really appreciate it.